python/cucim/src/cucim/skimage/measure/_regionprops_gpu_misc_kernels.py

```diff
@@ -0,0 +1,721 @@
+        List of labeled regions that are less than 2 pixel gap from another
+        label. Used when robust=True. If not provided and robust=True, it
+        will be computed internally.
+    props_dict : dict or None, optional
+        Dictionary of pre-computed properties (e.g. "slice"). The output of this
+        function will be stored under key "perimeter" within this dictionary.
+    pixels_per_thread : int, optional
+        Number of pixels processed per thread on the GPU during the final
+        weighted summation.
+
+    Returns
+    -------
+    perimeter : float
+        Total perimeter of all objects in binary image.
+
+    Notes
+    -----
+    The `perimeter` method does not consider the boundary along the image edge
+    as image as part of the perimeter, while the `perimeter_crofton` method
+    does. In any case, an object touching the image edge likely extends outside
+    of the field of view, so an accurate perimeter cannot be measured for such
+    objects.
+
+    If the labeled regions have holes, the hole edges will be included in this
+    measurement. If this is not desired, use regionprops_label_filled to fill
+    the holes and then pass the filled labels image to this function.
+
+    References
+    ----------
+    .. [1] K. Benkrid, D. Crookes. Design and FPGA Implementation of
+           a Perimeter Estimator. The Queen's University of Belfast.
+           http://www.cs.qub.ac.uk/~d.crookes/webpubs/papers/perimeter.doc
+
+    See Also
+    --------
+    perimeter_crofton
+
+    Examples
+    --------
+    >>> import cupy as cp
+    >>> from skimage import data
+    >>> from cucim.skimage import util
+    >>> from cucim.skimage.measure import label
+    >>> # coins image (binary)
+    >>> img_coins = cp.array(data.coins() > 110)
+    >>> # total perimeter of all objects in the image
+    >>> perimeter(img_coins, neighborhood=4)  # doctest: +ELLIPSIS
+    array(7796.86799644)
+    >>> perimeter(img_coins, neighborhood=8)  # doctest: +ELLIPSIS
+    array(8806.26807333)
+    """
+    if max_label is None:
+        max_label = int(labels.max())
+
+    binary_image = labels > 0
+    if robust and labels_close is None:
+        labels_close = _find_close_labels(labels, binary_image, max_label)
+    if neighborhood == 4:
+        footprint = cp.array([[0, 1, 0], [1, 1, 1], [0, 1, 0]], dtype=cp.uint8)
+    else:
+        footprint = 3
+
+    eroded_image = ndi.binary_erosion(binary_image, footprint, border_value=0)
+    border_image = binary_image.view(cp.uint8) - eroded_image
+
+    perimeter_weights, perimeter_coefs = _get_perimeter_weights_and_coefs(
+        cp.float32
+    )
+
+    perimeter_image = ndi.convolve(
+        border_image,
+        perimeter_weights,
+        mode="constant",
+        cval=0,
+        output=cp.uint8,
+    )
+
+    min_integer_type = _get_min_integer_dtype(max_label, signed=False)
+    # if labels.dtype != min_integer_type:
+    #     labels = labels_dilated.astype(min_integer_type)
```

Comment:
We may want to remove this or move these lines after line 224 (after `labels_dilated` is defined).